xbmc/filesystem/File.cpp

```diff
@@ -261,39 +258,32 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
   try
   {
     bool bPathInCache;
-
-    CURL url(URIUtils::SubstitutePath(file)), url2(url);
```

Comment:
save resources, save time. There is no need to make 2 copies od URL (beside all the things happen in the constructor (Concatting strings from a, barsing this string in b)